src/math/deconvolution_2d.jl

```diff
@@ -0,0 +1,1069 @@
+### A Pluto.jl notebook ###
+# v0.19.25
+
+using Markdown
+using InteractiveUtils
+
+# This Pluto notebook uses @bind for interactivity. When running this notebook outside of Pluto, the following 'mock version' of @bind gives bound variables a default value (instead of an error).
+macro bind(def, element)
+    quote
+        local iv = try Base.loaded_modules[Base.PkgId(Base.UUID("6e696c72-6542-2067-7265-42206c756150"), "AbstractPlutoDingetjes")].Bonds.initial_value catch; b -> missing; end
+        local el = $(esc(element))
+        global $(esc(def)) = Core.applicable(Base.get, el) ? Base.get(el) : iv(el)
+        el
+    end
+end
+
+# ╔═╡ 746ff659-88ab-4ff4-8cba-43c798cacd3e
+using Images,TestImages, PlutoUI
+
+# ╔═╡ 04a5c6d4-f8d5-11ed-141a-35481b811ee9
+md"""
+## Images as Lists of numbers
+Hi There! Remember when you were kid and you used to play with legos ? you probably would put tiny pieces of different colors to form different shapes and elements ? 
+
+Well images in computers work exactly the same way! Each image is made of tiny elements we call pixels. Since our computers only understand numbers, these pixels are given to the computer as a list of numbers.  
```

Comment:
I would probably state here what the notebook is about, something like: `in the following we will explore [..]`

Reply:
maybe you can mention filtering already here as well? And explain it intuitively? 

> we will also introduce filtering, which is a mathematical operation to modify images by smoothing, highlighting etc

or something like that